### PR TITLE
read colors from template

### LINKED ISSUE #70: Plotly Express does not play well with templates
Hi,

I noticed that Plotly Express is quite opinionated with some default values, which unfortunately do not play well with template's defined values such as graph height or color scale.

For example, I would expect `px.histogram(some_dataframe, template="seaborn")` to use the seaborn color palette.

Is there any plans to pick up default values in the active template instead?

### PR DESCRIPTION
**Pull request by nicolaskruchten**
closes #70

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
This PR now has `px` defer to the default template if it's not `None`, and reads the colors from the template.

File: plotly_express/_core.py [hunk 1]
```diff
@@ -597,6 +599,51 @@ def infer_config(args, constructor, trace_patch):
...
+        if pio.templates.default is not None:
+            args["template"] = pio.templates.default
+        else:
+            args["template"] = "plotly"
+
+    # retrieve the actual template if we were given a name
+    try:
+        template = pio.templates[args["template"]]
+    except Exception:
+        template = args["template"]
```
Comment (jonmmease) [inline plotly_express/_core.py]: Is the idea here to catch a `go.layout.Template` object being passed in directly?

Comment (nicolaskruchten) [reply plotly_express/_core.py]: Yeah

**Comment by nicolaskruchten** [comment]
Thanks!

This is a case where tests will be useful :)

I’ll add an issue to unit test this function.

**Pull request merged**